keps/sig-node/4210-max-image-gc-age/README.md

```diff
@@ -0,0 +1,368 @@
+- [ ] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input (including test refactors)
+  - [ ] e2e Tests for all Beta API Operations (endpoints)
+  - [ ] (R) Ensure GA e2e tests meet requirements for [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+  - [ ] (R) Minimum Two Week Window for GA e2e tests to prove flake free
+- [ ] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+Add an option ImageMaximumGCAge, which allows an admin to specify a time after which unused images will be garbage collected
+by the Kubelet, regardless of disk usage.
+
+## Motivation
+
+Currently, all image garbage collection the Kubelet is triggered by disk usage going over a threshold (ImageGCLowThresholdPercent).
+However, there are cases that additional conditions could be considered useful. One such condition is maximum age of an image.
+If an image is unused for a long time (the exact amount of time will be decided, but on the order of weeks is what comes to mind),
+then it is not likely to be used again.
+
+One such condition that can be imagined are clusters with automatic upgrades. If a cluster goes through an upgrade process,
+it will likely have images cached from the old release (old kube-apiserver/etcd/etc). While these images would eventually get removed
+through the disk usage condition, they will needlessly occupy disk space before that.
+
+### Goals
+
+- Introduce an option to the Kubelet ImageMaximumGCAge
+
+### Non-Goals
+
+- Introduce other conditions for image garbage collection to trigger.
+    - A WG was put together in SIG-Node to collect other GC use cases, and no other conditions were identified.
+
+## Proposal
+
+Kubelet has three different configuration fields for image garbage collection:
+- ImageMinimumGCAge: the youngest an image can be to be qualified for garbage collection
```

Comment:
I was wondering whether ImageMinimumGCAge can be used for ImageGCMaximumAge use cases. Once the image passes ImageGCMaximumAge, they can be GCed?

Reply:
I had thought about this, but ImageGCMinimumAge is only relevant when the node is under disk pressure, and is a condition for disqualifying images from disk pressure cleaning. ImageGCMaximumAge seeks to qualify otherwise unqualified images as it will trigger before disk pressure

Reply:
> but ImageGCMinimumAge is only relevant when the node is under disk pressure

Can we remove this condition instead since ImageGCMinimumAge indicates the minimum life cycle of  the image ?

A sweeping process can clean up them as long as ImageGCMinimumAge satisfied.  The idea came from past clustering experience, but not kubernetes though.